lib/kubernetes-deploy/ejson_secret_provisioner.rb

```diff
@@ -31,7 +32,11 @@ def secret_changes_required?
 
     def run
       create_secrets
-      prune_managed_secrets
+      if @prune
+        prune_managed_secrets
+      else
+        @logger.summary.add_action("no secrets were pruned")
```

Comment:
IMO this isn't necessary; we don't usually record things we _didn't_ do, and there's no similar log when we skip pruning of regular resources. Also, this is just reporting the setting; it's possible there was nothing to prune in any case. If you feel we should log something about pruning disabled (could be useful, but I don't feel strongly either way), I think it should be logged in the "initializing deploy" section.

Reply:
I'll remove that and update the test